jumanji/environments/swarms/search_and_rescue/env.py

```diff
@@ -0,0 +1,395 @@
+    AgentAndAllTargetObservationFn,
+    ObservationFn,
+)
+from jumanji.environments.swarms.search_and_rescue.reward import RewardFn, SharedRewardFn
+from jumanji.environments.swarms.search_and_rescue.types import Observation, State, TargetState
+from jumanji.environments.swarms.search_and_rescue.viewer import SearchAndRescueViewer
+from jumanji.types import TimeStep, restart, termination, transition
+from jumanji.viewer import Viewer
+
+
+class SearchAndRescue(Environment):
+    """A multi-agent search environment
+
+    Environment modelling a collection of agents collectively searching
+    for a set of targets on a 2d environment. Agents are rewarded
+    (individually) for coming within a fixed range of a target that has
+    not already been detected. Agents visualise their local environment
+    (i.e. the location of other agents and targets) via a simple segmented
+    view model. The environment area is a uniform square space with wrapped
+    boundaries.
+
+    An episode will terminate if all targets have been located by the team of
+    searching agents.
+
+    - observation: `Observation`
+        searcher_views: jax array (float) of shape (num_searchers, channels, num_vision)
+            Individual local views of positions of other agents and targets, where
+            channels can be used to differentiate between agents and targets.
+            Each entry in the view indicates the distant to another agent/target
+            along a ray from the agent, and is -1.0 if nothing is in range along the ray.
+            The view model can be customised using an `ObservationFn` implementation, e.g.
+            the view can include agents and all targets, agents and found targets,or
+            just other agents.
+        targets_remaining: (float) Number of targets remaining to be found from
+            the total scaled to the range [0, 1] (i.e. a value of 1.0 indicates
+            all the targets are still to be found).
+        step: (int) current simulation step.
+        positions: jax array (float) of shape (num_searchers, 2) search agent positions.
+
+    - action: jax array (float) of shape (num_searchers, 2)
+        Array of individual agent actions. Each agents actions rotate and
+        accelerate/decelerate the agent as [rotation, acceleration] on the range
+        [-1, 1]. These values are then scaled to update agent velocities within
+        given parameters (i.e. a value of -+1 is the maximum acceleration/rotation).
+
+    - reward: jax array (float) of shape (num_searchers,)
+        Arrays of individual agent rewards. A reward of +1 is granted when an agent
+        comes into contact range with a target that has not yet been found, and
+        the target is within the searchers view cone. Rewards can be shared
+        between agents if a target is simultaneously detected by multiple agents,
+        or each can be provided the full reward individually.
```

Comment:
Can you add a similar sentence here about customizing the reward with `RewardFn`s same as you did for the observation